src/Tizen.NUI.Components/Controls/AlertDialog.cs

```diff
@@ -190,6 +190,12 @@ private string InternalTitle
                 {
                     textLabel.Text = title;
                 }
+
+                if (Accessibility.Accessibility.IsEnabled && IsHighlighted)
+                {
+                    AccessibilityName = title;
+                    EmitAccessibilityEvent(AccessibilityPropertyChangeEvent.Name);
```

Comment:
It is not necessary to check `Accessibility.IsEnabled` before calling `Emit...`, all event emissions are guarded for `IsEnabled` (i.e. `IsUp` in DALi syntax, see `BridgeObject::Emit*()` in `DALI_ADAPTOR/dali/internal/accessibility/bridge/bridge-object.cpp`). This was introduced specifically so that DALi/NUI code would not have to check `IsEnabled` before every `Emit...`.


```c++
void BridgeObject::Emit(Accessible* obj, ObjectPropertyChangeEvent event)
{
  // ...
  
  if(!IsUp() || obj->IsHidden() || obj->GetSuppressedEvents()[AtspiEvent::PROPERTY_CHANGED])
  {
    return;
  }
```

Reply:
Oh.. I got it.
I just copied other codes in Button.cs, Progress.cs, Slider.cs etc. Are they unnecessary now?
Then, I'll remove them together.
For example,
```C#
// Button.cs
                if (Accessibility.Accessibility.IsEnabled && IsHighlighted && String.IsNullOrEmpty(AccessibilityName) && GetAccessibilityNameSignal().Empty())
                {
                    EmitAccessibilityEvent(AccessibilityPropertyChangeEvent.Name);
                }
```

```C#
// Progress.cs
                if (Accessibility.Accessibility.IsEnabled && IsHighlighted)
                {
                    EmitAccessibilityEvent(AccessibilityPropertyChangeEvent.Value);
                }
```

Reply:
BTW, I'll change other classes like `AccessibilityGetName()`, etc.
Then, I don't need to add `EmitAccessibilityEvent(AccessibilityPropertyChangeEvent.Name)` separately, do I?

Reply:
Checking `Accessibility.IsEnabled` is unnecessary before `EmitAccessibilityEvent` (as already explained, `Emit` checks `IsEnabled` before sending anything over D-Bus). It looks like the existing NUI code predates this DALi patch: https://review.tizen.org/gerrit/#/c/platform/core/uifw/dali-adaptor/+/268774/, so it can be simplified now to take advantage of said patch.

However, it really depends on the UX guide and the expected TTS feedback whether to `EmitAccessibilityEvent` and where and when to do so. If (for the currently highlighted control) it is expected to hear the new name / description / value when any of these change, there should be an `EmitAccessibilityEvent` (as can be seen for `Button` and `Progress`). For `Progress` the UX guide specifies that the new value should be read aloud when it changes (and that's why we `Emit...(Value)` there), similarly, `Button` changing its title on click is also expected to cause TTS feedback. But in the general case, not every change of name should be propagated to the Screen Reader in order not to spam the user with TTS feedback (an alert popup title with countdown timer in milliseconds is an extreme example of this).

Reply:
Now I understood. Thanks.